src/idl_gen_js.cpp

```diff
@@ -785,4 +1032,28 @@ std::string JSMakeRule(const Parser &parser,
   return make_rule;
 }
 
+bool GenerateJS(const Parser &parser, const std::string &path,
+                const std::string &file_name) {
+  js::JsGenerator generator(parser, path, file_name, false);
+  return generator.generate();
+}
+
+std::string JSMakeRule(const Parser &parser,
+                       const std::string &path,
+                       const std::string &file_name) {
+  return MakeRule(parser, path, file_name, false);
+}
+
+bool GenerateTS(const Parser &parser, const std::string &path,
+                const std::string &file_name) {
+  js::JsGenerator generator(parser, path, file_name, true);
+  return generator.generate();
+}
+
+std::string TSMakeRule(const Parser &parser,
+                       const std::string &path,
+                       const std::string &file_name) {
+  return MakeRule(parser, path, file_name, true);
+}
+
```

Comment:
With `JsLangParameters`, these can be deduplicated, here's [one way it can be done](https://github.com/fasterthanlime/flatbuffers/blob/c2a3f43ee80db2c9a5b14503b165b03597f9f6f9/src/idl_gen_js.cpp#L978-L1000).